docs/markdown/Using Pants/remote-caching-execution/remote-execution.md

```diff
@@ -91,6 +91,9 @@ remote_execution_address = "grpcs://build.example.com:443"
 remote_instance_name = "main"
 # This is optional, Pants will auto-discover certificates otherwise.
 remote_ca_certs_path = "/etc/ssl/certs/ca-certificates.crt"
+# this allows you to setup mTLS with a client certificate and key.
+remote_mtls_certs_path = "/etc/ssl/certs/client-cert.pem"
```

Comment:
Naming questions:
1. Can there be multiple certificates in the file at that path or just one? I.e., should the option be `remote_mtls_cert_path` instead of `remote_mtls_certs_path`?
2. These options are really only about enabling the TLS client certificate and key. mTLS actually implies the superset of doing _mutual_ client cert verification and server cert verification. Should we rename these options to reflect that they only pertain to the client certificate? Bazel calls its equivalent options [`--tls_client_certificate`](https://bazel.build/reference/command-line-reference#flag--tls_client_certificate) and [`--tls_client_key`](https://bazel.build/reference/command-line-reference#flag--tls_client_key). Maybe we s should similarly call the Pants options `remote_tls_client_cert_path` and `remote_tls_client_key`?

Reply:
cc @stuhood and @benjyw on input on the option naming

Reply:
1) Yes, I think it's always a chain formally.

2) Makes sense on naming, I had that during my initial setup but since the internal config calls it mtls and Pants is never a server I went for that name. But I'm happy to rename.

Reply:
Probably worth asking for some opinions in Slack. I'll tag this comment thread there.

Reply:
Also given the mTLS code was never visible to users, my view is we are free to name it as if this were a new feature. :)

Reply:
Yeah, I think `client` in the name makes sense, presumably in an mTLS scenario the server cert still comes via the CA chain?

Reply:
Renamed all the public places that I introduced, and most validation messages. Kept the lower level structs as `mtls_data` to harmonize with `MtlsConfig`.

Reply:
Kept mTLS in the docs as a search needle, but also added `client authentication` so both are searchable.